internal/cache/file/downloader/job.go

```diff
@@ -124,6 +126,47 @@ func (job *Job) Cancel() {
 	defer job.mu.Unlock()
 }
 
+// addSubscriber adds subscriber for download job and returns channel which is
+// notified when the download is completed at least till the subscribed offset
+// or in case of failure.
+//
+// Not concurrency safe and requires LOCK(job.mu)
+func (job *Job) addSubscriber(subscribedOffset int64) (notificationC <-chan JobStatus) {
```

Comment:
Ideally, we should give a jobSubscriber as an input Parameter to addSubscriber. And accordingly we should not return anything. What do you think?

Reply:
Would it work if we change the name from addSubscriber to subscribe and keep input and return types same ?

Reply:
Honestly speaking, I am more inclined towards addSubscriber(jobSubscriber). In this way we give flexibility to client  to give their notificationChannel.

Reply:
We decided both the options are fine. Decided to go with subscribe option as the other one increase code for test.